types/src/transaction/metadata.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This file implements Libra transaction metadata types to allow
+//! easy parsing and introspection into metadata, whether the transaction
+//! is using regular subaddressing, is subject to travel rule or corresponds
+//! to an on-chain payment refund.
+
+use serde::{Deserialize, Serialize};
+
+/// List of all supported metadata types
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum MetadataType {
```

Comment:
I would suggest to remove the `-Type` suffix here (seems like a typo?) and below (re-using names between variants and containers is well supported).

Reply:
Hmm I don't think that's a typo, in general you type opaque data structures with:
1] type
2] struct details

But given that rust supports storing struct details in each variant, I'm fine with it.

Reply:
Actually the ones below are typos